ReSharper.FSharp/src/FSharp.Psi/src/Util/SolutionExtension.cs

```diff
@@ -0,0 +1,19 @@
+using JetBrains.Metadata.Reader.API;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.FSharp.Shim.TypeProviders;
+using JetBrains.ReSharper.Plugins.FSharp.TypeProviders.Protocol.Models;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+{
+  public static class SolutionExtension
+  {
+    public static bool TryGetProvidedType(this ISolution solution, IClrTypeName name,
+      out ProxyProvidedTypeWithContext providedType)
+    {
+      providedType = null;
+      var extensionTypingProvider = solution.GetComponent<IProxyExtensionTypingProvider>();
+      return extensionTypingProvider.TypeProvidersManager?.Context.ProvidedAbbreviations.TryGetValue(name.FullName,
```

Comment:
What is two different projects provide different but the same-abbreviated types? Shouldn't it take a psi module into account?

Reply:
Yes, we have to take this into account, I'm working on it now.